larq_zoo/data.py

```diff
@@ -31,12 +32,34 @@ def preprocess_input(image):
     return result
 
 
-@registry.register_preprocess("imagenet2012", (IMAGE_SIZE, IMAGE_SIZE, 3))
-@registry.register_preprocess("oxford_flowers102", (IMAGE_SIZE, IMAGE_SIZE, 3))
-def default(image_bytes, training):
-    return preprocess_image_bytes(
-        image_bytes, is_training=training, image_size=IMAGE_SIZE
-    )
+class ImageClassification(Preprocessing):
+    @property
+    def kwargs(self):
+        return {
+            "input_shape": self.features["image"].shape,
+            "num_classes": self.features["label"].num_classes,
+        }
+
+    def inputs(self, data, training):
+        return tf.cast(data["image"], tf.float32)
```

Comment:
Why is the training argument in this function?

Reply:
The `inputs` and `outputs` functions here overwrite those functions form the [`Preprocessing`](https://github.com/larq/zookeeper/blob/1210127673e3c4a9bd1ec1bcdc4fc6200b16797d/zookeeper/preprocessing.py#L38) class which has the `training` argument. The input function is overwritten below where it uses the argument. I included it here to keep the method signature the same for clarity.